packages/shared/src/api/dex.ts

```diff
@@ -0,0 +1,57 @@
+import { ChainKind, ChainNames, ChainTypes } from '@/enums'
+import { isUndefined } from '@/helpers'
+import type {
+  BridgeChain,
+  HexString,
+  InternalAccountBalance,
+  InternalBridgeChain,
+  InternalSupportedToken,
+} from '@/types'
+
+import { dexApi, loadDataViaLoop } from './api'
+
+export const loadSupportedChains = async ({
+  type,
+  kind,
+}: { type?: ChainTypes; kind?: ChainKind } = {}): Promise<BridgeChain[]> => {
+  const { data } = await dexApi.get<InternalBridgeChain[]>('/chains', {
+    query: {
+      ...(isUndefined(type) ? {} : { 'filter[type]': type }),
+      ...(kind ? { 'filter[kind]': kind } : {}),
```

Comment:
can be simplified to `kind && { 'filter...`, you can use null/undefined with object spreading